src/main/java/Asuka.java

```diff
@@ -0,0 +1,256 @@
+        }
+        System.out.println("____________________________________________________________");
+    }
+
+    private static void unmark(String[] inputs, int taskCount, Task[] tasks) {
+        int taskNumber;
+        taskNumber = Integer.parseInt(inputs[1]);
+        System.out.println("____________________________________________________________");
+        if (taskNumber > 0 && taskNumber <= taskCount) {
+
+            System.out.println("Nice! I've marked this task as undone: ");
+            tasks[taskNumber - 1].markAsUndone();
+            tasks[taskNumber - 1].printTask();
+        }
+        else {
+            System.out.println("Task index out of bound, the total number of task(s) you have now is: " + taskCount);
+        }
+        System.out.println("____________________________________________________________");
+    }
+
+    private static void mark(String[] inputs, int taskCount, Task[] tasks) {
+        int taskNumber;
+        taskNumber = Integer.parseInt(inputs[1]);
+        System.out.println("____________________________________________________________");
+        if (taskNumber > 0 && taskNumber <= taskCount) {
+            System.out.println("Nice! I've marked this task as done: ");
+            tasks[taskNumber - 1].markAsDone();
+            tasks[taskNumber - 1].printTask();
+        }
+        else {
+            System.out.println("Task index out of bound, the total number of task(s) you have now is: " + taskCount);
+        }
+        System.out.println("____________________________________________________________");
+    }
+
+    private static Commands getCommands(String input, String[] inputs) {
+        if ((inputs[0].equalsIgnoreCase("todo")) || (inputs[0].equalsIgnoreCase("deadline")) || (inputs[0].equalsIgnoreCase("event"))){
+            if (inputs.length == 1)
+            {
+                return null;
+            }
+            return Commands.add;
+        }
+        if (inputs.length > 2) {
+            return null;
+        } else if (inputs.length == 2 && !inputs[0].equalsIgnoreCase("mark") && !inputs[0].equalsIgnoreCase("unmark")) {
+            return null;
+        }
+        if (inputs.length == 2)
+        {
+            if (inputs[0].equalsIgnoreCase("mark") && Integer.parseInt(inputs[1]) <= MAX_TASKS && Integer.parseInt(inputs[1]) > 0) {
+                return Commands.mark;
+            } else if (inputs[0].equalsIgnoreCase("unmark") && Integer.parseInt(inputs[1]) <= MAX_TASKS && Integer.parseInt(inputs[1]) > 0) {
+                return Commands.unmark;
+            } else {
+                return null;
+            }
+        }
+        if (input.equalsIgnoreCase("bye")) {
+            return Commands.bye;
+        } else if (input.equalsIgnoreCase("list")) {
+            return Commands.list;
+        }
+        return null;
+    }
+
+    private static void printWelcomeMessage() {
+        System.out.println("____________________________________________________________");
```

Comment:
Good work on creating a method for welcome message, makes Main statement neater and more readable.